azurerm/helpers/azure/app_service.go

```diff
@@ -1114,16 +1142,55 @@ func FlattenAppServiceLogs(input *web.SiteLogsConfigProperties) []interface{} {
 				blobStorageItem["retention_in_days"] = *blobStorageInput.RetentionInDays
 			}
 
-			blobStorage = append(blobStorage, blobStorageItem)
+			// The API returns a non nil application logs object when other logs are specified so we'll check that this structure is empty before adding it to the statefile.
+			if blobStorageInput.SasURL != nil {
+				blobStorage = append(blobStorage, blobStorageItem)
+
+				appLogsItem["azure_blob_storage"] = blobStorage
+			}
 
-			appLogsItem["azure_blob_storage"] = blobStorage
 		}
 
-		appLogs = append(appLogs, appLogsItem)
+		if len(appLogsItem) > 0 {
+			appLogs = append(appLogs, appLogsItem)
+		}
 
 		result["application_logs"] = appLogs
 	}
 
+	if input.HTTPLogs != nil {
+		httpLogs := make([]interface{}, 0)
+
+		httpLogsItem := make(map[string]interface{})
+
+		if fileSystemInput := input.HTTPLogs.FileSystem; fileSystemInput != nil {
+			fileSystem := make([]interface{}, 0)
+
+			fileSystemItem := make(map[string]interface{})
+
+			if fileSystemInput.RetentionInDays != nil {
+				fileSystemItem["retention_in_days"] = *fileSystemInput.RetentionInDays
+			}
+
+			if fileSystemInput.RetentionInMb != nil {
+				fileSystemItem["retention_in_mb"] = *fileSystemInput.RetentionInMb
+			}
+
+			// The API returns a non nil filesystem logs object when other logs are specified so we'll check that this is disabled before adding it to the statefile.
```

Comment:
is this supposed to be returns a nil logs object?

Reply:
I'm not sure! It's returning an empty object that I believe should be nil